sdk/storage/azblob/zt_examples_test.go

```diff
@@ -124,15 +118,325 @@ func Example() {
+	parallelism := 3
+	downloadOffset := 0
+	downloadCount := azblob.CountToEnd
+	_, err = serviceClient.DownloadFile(context.TODO(), containerName, blobName, destFile,
+		&blob.DownloadFileOptions{
+			Count:       int64(downloadCount),
+			Offset:      int64(downloadOffset),
+			BlockSize:   int64(blockSize),
+			Parallelism: uint16(parallelism),
+			// If Progress is non-nil, this function is called periodically as bytes are uploaded.
+			Progress: func(bytesTransferred int64) {
+				fmt.Println(bytesTransferred)
+			},
+		})
+
+	// Assert download was successful
+	handleError(err)
+
+	fileSize := 8 * 1024 * 1024 // assuming blob size is 8MB.
+
+	// Assert downloaded data is consistent
+	var destBuffer []byte
+	if downloadCount == azblob.CountToEnd {
+		destBuffer = make([]byte, fileSize-downloadOffset)
+	} else {
+		destBuffer = make([]byte, downloadCount)
+	}
+
+	_, err = destFile.Read(destBuffer)
+	handleError(err)
+
+}
+
+func Example_client_NewListBlobsPager() {
+	accountName, ok := os.LookupEnv("AZURE_STORAGE_ACCOUNT_NAME")
+	if !ok {
+		panic("AZURE_STORAGE_ACCOUNT_NAME could not be found")
+	}
+	serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	handleError(err)
+
+	serviceClient, err := azblob.NewClient(serviceURL, cred, nil)
+	handleError(err)
+
+	pager := serviceClient.NewListBlobsPager("testcontainer", &azblob.ListBlobsOptions{
+		Include: []azblob.ListBlobsIncludeItem{azblob.ListBlobsIncludeItemVersions, azblob.ListBlobsIncludeItemDeleted},
+	})
+
+	for pager.More() {
+		resp, err := pager.NextPage(ctx)
+		handleError(err) // if err is not nil, break the loop.
+		for _, _blob := range resp.Segment.BlobItems {
+			fmt.Printf("%v", _blob)
+		}
+	}
+}
+
+func Example_client_DeleteBlob() {
+	accountName, ok := os.LookupEnv("AZURE_STORAGE_ACCOUNT_NAME")
+	if !ok {
+		panic("AZURE_STORAGE_ACCOUNT_NAME could not be found")
+	}
+	serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	handleError(err)
+
+	serviceClient, err := azblob.NewClient(serviceURL, cred, nil)
+	handleError(err)
+
+	resp, err := serviceClient.DeleteBlob(context.TODO(), "testcontainer", "testblob", nil)
+	handleError(err)
+	fmt.Println(resp)
+}
+
+func Example_client_UploadStream() {
+	accountName, ok := os.LookupEnv("AZURE_STORAGE_ACCOUNT_NAME")
+	if !ok {
+		panic("AZURE_STORAGE_ACCOUNT_NAME could not be found")
+	}
+	serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	handleError(err)
+
+	serviceClient, err := azblob.NewClient(serviceURL, cred, nil)
+	handleError(err)
+
+	// Set up test blob
+	containerName := "testcontainer"
+	bufferSize := 8 * 1024 * 1024
+	blobName := "test_upload_stream.bin"
+	blobData := make([]byte, bufferSize)
+	blobContentReader := streaming.NopCloser(bytes.NewReader(blobData))
```

Comment:
Shouldn't need the NopCloser

Reply:
Yeah. bytes.NewReader anyway is returning `*Reader` back to us.
Removed.